corgi/tasks/rhel_compose.py

```diff
@@ -28,39 +28,41 @@ def save_compose(stream_name) -> None:
         compose_id, compose_created_date, compose_data = RhelCompose.fetch_compose_data(
             compose_url, variants
         )
-        if "srpms" in compose_data:
-            srpms = compose_data["srpms"].keys()
-            find_build_id_calls = _brew_srpm_lookup(brew, srpms)
-            for srpm, call in find_build_id_calls:
-                build_id = call.result
-                if not build_id:
-                    for filename in compose_data["srpms"][srpm]:
-                        logger.debug(
-                            "Didn't find build with NVR %s, using rpm filename: %s",
-                            srpm,
-                            filename,
-                        )
-                        rpm_data = brew.koji_session.getRPM(filename)
-                        if not rpm_data:
-                            # Try the next srpm rpm filename
-                            continue
-                        build_id = rpm_data["build_id"]
-                        # found the build_id, stop iterating filenames
-                        break
-                _, created = ProductComponentRelation.objects.get_or_create(
-                    external_system_id=compose_id,
-                    product_ref=stream_name,
-                    build_id=build_id,
-                    defaults={"type": ProductComponentRelation.Type.COMPOSE},
-                )
-                if created:
-                    no_of_relations += 1
+        srpms = compose_data.get("srpms", {}).keys()
+        if not srpms:
+            continue
+        find_build_id_calls = _brew_srpm_lookup(brew, srpms)
+        for srpm, call in find_build_id_calls:
+            build_id = call.result
+            if not build_id:
+                for filename in compose_data["srpms"][srpm]:
+                    logger.debug(
+                        "Didn't find build with NVR %s, using rpm filename: %s",
+                        srpm,
+                        filename,
+                    )
+                    rpm_data = brew.koji_session.getRPM(filename)
+                    if not rpm_data:
+                        # Try the next srpm rpm filename
+                        continue
+                    build_id = rpm_data["build_id"]
+                    # found the build_id, stop iterating filenames
+                    break
+            # TODO: What if build_id still not found?
```

Comment:
@jasinner This could happen when all the filenames have no rpm_data - in that case, should we raise an error? Right now we pass build_id=None to ProductComponentRelation.objects.get_or_create().

Reply:
If filenames have no rpm_data I guess the best we can do is log a warning about it and skip this SRPM.